Trigger/src/ReadTriggerInfo_module.cc

```diff
@@ -744,108 +744,88 @@ namespace mu2e {
-      }
-      //fill the Calo-Only Trigger bits info
-      findTrigIndex(_trigCaloOnly, moduleLabel, index);
-      if ( flag.hasAnyProperty(caloFlag) || flag.hasAnyProperty(caloTrigSeedFlag)){ 
-	_trigCaloOnly[index].label  = moduleLabel;
-	_trigCaloOnly[index].counts = _trigCaloOnly[index].counts + 1;
-	const CaloTrigSeed*clseed = trigInfo->caloTrigSeed().get();
-	if(clseed) {
-	  fillCaloTrigSeedInfo(index, clseed, _caloTSeedHist);
-	  fillOccupancyInfo   (_nTrackTrig*2+index, sdCol, cdCol, _occupancyHist);
-	}
-	trigFlag_index.push_back(index_all);
-      }
+	  //fill the Global Trigger bits info
+	  findTrigIndex(_trigAll, moduleLabel, index_all);
+	  _trigAll[index_all].label  = moduleLabel;
 
-      //fill the CaloCalib Trigger bits info
-      findTrigIndex(_trigCaloCalib, moduleLabel, index);
-      if ( flag.hasAnyProperty(caloCalibFlag)){ 
-	_trigCaloCalib[index].label  = moduleLabel;
-	_trigCaloCalib[index].counts = _trigCaloCalib[index].counts + 1;
-	const CaloCluster*cluster = trigInfo->caloCluster().get();
-	if(cluster) fillCaloCalibTrigInfo(index, cluster, _caloCalibHist);
-	trigFlag_index.push_back(index_all);
-      }
-      
-      //fill the Track Trigger bits info
-      findTrigIndex(_trigTrack, moduleLabel, index);
-      if ( flag.hasAnyProperty(trackFlag)){ 
-	_trigTrack[index].label  = moduleLabel;
-	_trigTrack[index].counts = _trigTrack[index].counts + 1;
-	const KalSeed*kseed = trigInfo->track().get();
-	if(kseed) {
-	  fillTrackTrigInfo(index, kseed, _trkHist);
-	  fillOccupancyInfo(index, sdCol, cdCol, _occupancyHist);
-	}
-	trigFlag_index.push_back(index_all);
-      }
-       //fill the Helix Trigger bits info
-      findTrigIndex(_trigHelix, moduleLabel, index);
-      if ( flag.hasAnyProperty(helixFlag)){ 
-	_trigHelix[index].label  = moduleLabel;
-	_trigHelix[index].counts = _trigHelix[index].counts + 1;
-	const HelixSeed*hseed = trigInfo->helix().get();
-	if(hseed) {
-	  fillHelixTrigInfo(index, hseed, _helHist);
-	  fillOccupancyInfo(_nTrackTrig+index, sdCol, cdCol, _occupancyHist);
-	}
-      }
-      
-      //fill the Event-Prescaler Trigger bits info
-      findTrigIndex(_trigEvtPS, moduleLabel, index);
-      if ( flag.hasAnyProperty(prescalerFlag)){ 
-	_trigEvtPS[index].label  = moduleLabel;
-	_trigEvtPS[index].counts = _trigEvtPS[index].counts + 1;
+	  event.getByLabel(moduleLabel, hTrigInfoH);
+	  if (hTrigInfoH.isValid()){
+	    trigInfo = hTrigInfoH.product();
+	  }
+	  if ( moduleLabel.find(std::string("HSFilter")) != std::string::npos) {
+	    findTrigIndex(_trigHelix, moduleLabel, index);
+	    _trigHelix[index].label  = moduleLabel;
+	    _trigHelix[index].counts = _trigHelix[index].counts + 1;
+	    const HelixSeed*hseed = trigInfo->helix().get();
+	    if(hseed) {
+	      fillHelixTrigInfo(index, hseed, _helHist);
+	      fillOccupancyInfo(_nTrackTrig+index, sdCol, cdCol, _occupancyHist);
+	    }
+	    
+	  }else if ( moduleLabel.find("TSFilter") != std::string::npos){
```

Comment:
This code compares to hard-coded strings: shouldn't those be fcl parameters?  or come from the trigger table?  I'm not sure of the architecture to know which.

Reply:
I followed the convention I have been using for all the current Track Triggers. I also don't like it, but I'm not sure that making it a fcl parameter would be better, because the code is extremely tight to the output object of each stage.